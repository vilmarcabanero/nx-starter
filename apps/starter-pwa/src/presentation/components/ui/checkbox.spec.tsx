import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { Checkbox } from './checkbox';

describe('Checkbox Component Snapshots', () => {
  describe('checkbox states', () => {
    it('renders unchecked checkbox', () => {
      const { container } = render(<Checkbox />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders checked checkbox', () => {
      const { container } = render(<Checkbox checked />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders indeterminate checkbox', () => {
      const { container } = render(<Checkbox checked="indeterminate" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders disabled unchecked checkbox', () => {
      const { container } = render(<Checkbox disabled />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders disabled checked checkbox', () => {
      const { container } = render(<Checkbox disabled checked />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('checkbox attributes', () => {
    it('renders checkbox with custom id', () => {
      const { container } = render(<Checkbox id="custom-checkbox" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders checkbox with name attribute', () => {
      const { container } = render(<Checkbox name="agreement" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders checkbox with value attribute', () => {
      const { container } = render(<Checkbox value="yes" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders required checkbox', () => {
      const { container } = render(<Checkbox required />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('customization', () => {
    it('renders checkbox with custom className', () => {
      const { container } = render(<Checkbox className="custom-checkbox" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders checkbox with aria-invalid', () => {
      const { container } = render(<Checkbox aria-invalid="true" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders checkbox with aria-label', () => {
      const { container } = render(<Checkbox aria-label="Accept terms" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders checkbox with aria-describedby', () => {
      const { container } = render(<Checkbox aria-describedby="help-text" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('checkbox combinations', () => {
    it('renders required checked checkbox with custom class', () => {
      const { container } = render(
        <Checkbox required checked className="form-checkbox" id="terms" />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders disabled checkbox with aria-label', () => {
      const { container } = render(
        <Checkbox disabled aria-label="Disabled option" />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders checkbox with all accessibility attributes', () => {
      const { container } = render(
        <Checkbox
          id="accessible-checkbox"
          aria-label="Accessible checkbox"
          aria-describedby="checkbox-help"
          aria-invalid="false"
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
