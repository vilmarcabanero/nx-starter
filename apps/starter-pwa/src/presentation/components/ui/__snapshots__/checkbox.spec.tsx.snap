// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkbox Component Snapshots > checkbox attributes > renders checkbox with custom id 1`] = `
<button
  aria-checked="false"
  class="peer border-input dark:bg-input/30 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground dark:data-[state=checked]:bg-primary data-[state=checked]:border-primary focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive size-4 shrink-0 rounded-[4px] border shadow-xs transition-shadow outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
  data-slot="checkbox"
  data-state="unchecked"
  id="custom-checkbox"
  role="checkbox"
  type="button"
  value="on"
/>
`;

exports[`Checkbox Component Snapshots > checkbox attributes > renders checkbox with name attribute 1`] = `
<button
  aria-checked="false"
  class="peer border-input dark:bg-input/30 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground dark:data-[state=checked]:bg-primary data-[state=checked]:border-primary focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive size-4 shrink-0 rounded-[4px] border shadow-xs transition-shadow outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
  data-slot="checkbox"
  data-state="unchecked"
  role="checkbox"
  type="button"
  value="on"
/>
`;

exports[`Checkbox Component Snapshots > checkbox attributes > renders checkbox with value attribute 1`] = `
<button
  aria-checked="false"
  class="peer border-input dark:bg-input/30 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground dark:data-[state=checked]:bg-primary data-[state=checked]:border-primary focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive size-4 shrink-0 rounded-[4px] border shadow-xs transition-shadow outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
  data-slot="checkbox"
  data-state="unchecked"
  role="checkbox"
  type="button"
  value="yes"
/>
`;

exports[`Checkbox Component Snapshots > checkbox attributes > renders required checkbox 1`] = `
<button
  aria-checked="false"
  aria-required="true"
  class="peer border-input dark:bg-input/30 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground dark:data-[state=checked]:bg-primary data-[state=checked]:border-primary focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive size-4 shrink-0 rounded-[4px] border shadow-xs transition-shadow outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
  data-slot="checkbox"
  data-state="unchecked"
  role="checkbox"
  type="button"
  value="on"
/>
`;

exports[`Checkbox Component Snapshots > checkbox combinations > renders checkbox with all accessibility attributes 1`] = `
<button
  aria-checked="false"
  aria-describedby="checkbox-help"
  aria-invalid="false"
  aria-label="Accessible checkbox"
  class="peer border-input dark:bg-input/30 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground dark:data-[state=checked]:bg-primary data-[state=checked]:border-primary focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive size-4 shrink-0 rounded-[4px] border shadow-xs transition-shadow outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
  data-slot="checkbox"
  data-state="unchecked"
  id="accessible-checkbox"
  role="checkbox"
  type="button"
  value="on"
/>
`;

exports[`Checkbox Component Snapshots > checkbox combinations > renders disabled checkbox with aria-label 1`] = `
<button
  aria-checked="false"
  aria-label="Disabled option"
  class="peer border-input dark:bg-input/30 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground dark:data-[state=checked]:bg-primary data-[state=checked]:border-primary focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive size-4 shrink-0 rounded-[4px] border shadow-xs transition-shadow outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
  data-disabled=""
  data-slot="checkbox"
  data-state="unchecked"
  disabled=""
  role="checkbox"
  type="button"
  value="on"
/>
`;

exports[`Checkbox Component Snapshots > checkbox combinations > renders required checked checkbox with custom class 1`] = `
<button
  aria-checked="true"
  aria-required="true"
  class="peer border-input dark:bg-input/30 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground dark:data-[state=checked]:bg-primary data-[state=checked]:border-primary focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive size-4 shrink-0 rounded-[4px] border shadow-xs transition-shadow outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50 form-checkbox"
  data-slot="checkbox"
  data-state="checked"
  id="terms"
  role="checkbox"
  type="button"
  value="on"
>
  <span
    class="flex items-center justify-center text-current transition-none"
    data-slot="checkbox-indicator"
    data-state="checked"
    style="pointer-events: none;"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-check size-3.5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 6 9 17l-5-5"
      />
    </svg>
  </span>
</button>
`;

exports[`Checkbox Component Snapshots > checkbox states > renders checked checkbox 1`] = `
<button
  aria-checked="true"
  class="peer border-input dark:bg-input/30 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground dark:data-[state=checked]:bg-primary data-[state=checked]:border-primary focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive size-4 shrink-0 rounded-[4px] border shadow-xs transition-shadow outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
  data-slot="checkbox"
  data-state="checked"
  role="checkbox"
  type="button"
  value="on"
>
  <span
    class="flex items-center justify-center text-current transition-none"
    data-slot="checkbox-indicator"
    data-state="checked"
    style="pointer-events: none;"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-check size-3.5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 6 9 17l-5-5"
      />
    </svg>
  </span>
</button>
`;

exports[`Checkbox Component Snapshots > checkbox states > renders disabled checked checkbox 1`] = `
<button
  aria-checked="true"
  class="peer border-input dark:bg-input/30 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground dark:data-[state=checked]:bg-primary data-[state=checked]:border-primary focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive size-4 shrink-0 rounded-[4px] border shadow-xs transition-shadow outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
  data-disabled=""
  data-slot="checkbox"
  data-state="checked"
  disabled=""
  role="checkbox"
  type="button"
  value="on"
>
  <span
    class="flex items-center justify-center text-current transition-none"
    data-disabled=""
    data-slot="checkbox-indicator"
    data-state="checked"
    style="pointer-events: none;"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-check size-3.5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 6 9 17l-5-5"
      />
    </svg>
  </span>
</button>
`;

exports[`Checkbox Component Snapshots > checkbox states > renders disabled unchecked checkbox 1`] = `
<button
  aria-checked="false"
  class="peer border-input dark:bg-input/30 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground dark:data-[state=checked]:bg-primary data-[state=checked]:border-primary focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive size-4 shrink-0 rounded-[4px] border shadow-xs transition-shadow outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
  data-disabled=""
  data-slot="checkbox"
  data-state="unchecked"
  disabled=""
  role="checkbox"
  type="button"
  value="on"
/>
`;

exports[`Checkbox Component Snapshots > checkbox states > renders indeterminate checkbox 1`] = `
<button
  aria-checked="mixed"
  class="peer border-input dark:bg-input/30 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground dark:data-[state=checked]:bg-primary data-[state=checked]:border-primary focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive size-4 shrink-0 rounded-[4px] border shadow-xs transition-shadow outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
  data-slot="checkbox"
  data-state="indeterminate"
  role="checkbox"
  type="button"
  value="on"
>
  <span
    class="flex items-center justify-center text-current transition-none"
    data-slot="checkbox-indicator"
    data-state="indeterminate"
    style="pointer-events: none;"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-check size-3.5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 6 9 17l-5-5"
      />
    </svg>
  </span>
</button>
`;

exports[`Checkbox Component Snapshots > checkbox states > renders unchecked checkbox 1`] = `
<button
  aria-checked="false"
  class="peer border-input dark:bg-input/30 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground dark:data-[state=checked]:bg-primary data-[state=checked]:border-primary focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive size-4 shrink-0 rounded-[4px] border shadow-xs transition-shadow outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
  data-slot="checkbox"
  data-state="unchecked"
  role="checkbox"
  type="button"
  value="on"
/>
`;

exports[`Checkbox Component Snapshots > customization > renders checkbox with aria-describedby 1`] = `
<button
  aria-checked="false"
  aria-describedby="help-text"
  class="peer border-input dark:bg-input/30 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground dark:data-[state=checked]:bg-primary data-[state=checked]:border-primary focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive size-4 shrink-0 rounded-[4px] border shadow-xs transition-shadow outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
  data-slot="checkbox"
  data-state="unchecked"
  role="checkbox"
  type="button"
  value="on"
/>
`;

exports[`Checkbox Component Snapshots > customization > renders checkbox with aria-invalid 1`] = `
<button
  aria-checked="false"
  aria-invalid="true"
  class="peer border-input dark:bg-input/30 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground dark:data-[state=checked]:bg-primary data-[state=checked]:border-primary focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive size-4 shrink-0 rounded-[4px] border shadow-xs transition-shadow outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
  data-slot="checkbox"
  data-state="unchecked"
  role="checkbox"
  type="button"
  value="on"
/>
`;

exports[`Checkbox Component Snapshots > customization > renders checkbox with aria-label 1`] = `
<button
  aria-checked="false"
  aria-label="Accept terms"
  class="peer border-input dark:bg-input/30 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground dark:data-[state=checked]:bg-primary data-[state=checked]:border-primary focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive size-4 shrink-0 rounded-[4px] border shadow-xs transition-shadow outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
  data-slot="checkbox"
  data-state="unchecked"
  role="checkbox"
  type="button"
  value="on"
/>
`;

exports[`Checkbox Component Snapshots > customization > renders checkbox with custom className 1`] = `
<button
  aria-checked="false"
  class="peer border-input dark:bg-input/30 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground dark:data-[state=checked]:bg-primary data-[state=checked]:border-primary focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive size-4 shrink-0 rounded-[4px] border shadow-xs transition-shadow outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50 custom-checkbox"
  data-slot="checkbox"
  data-state="unchecked"
  role="checkbox"
  type="button"
  value="on"
/>
`;
